testutils/base.go

```diff
@@ -24,22 +24,20 @@ import (
 	"github.com/cockroachdb/cockroach/security"
 )
 
-// NewTestBaseContext creates a base context for testing.
+// NewRootTestBaseContext creates a base context for testing.
 // This uses embedded certs and the "root" user (default client user).
-// The certs file loader is overriden in individual main_test files.
-// This is meant to be used by external clients (as opposed to nodes).
-func NewTestBaseContext() *base.Context {
+// The "root" user has client certificates only.
+func NewRootTestBaseContext() *base.Context {
```

Comment:
The tests which grab the vars and change them should really be changed to making a copy. Making a copy would be easier if the tests were passing structs instead of pointers.

Generally, we should be passing structs more often than we are. If the method is taking a copy of the parameter or not making any modifications that need to be reflected back to the caller and we don't need to distinguish `nil`, then a struct should be preferred.